### PR TITLE
feat (api, cli): export application

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

File: sdk/application.go [hunk 1]
```diff
@@ -95,18 +96,58 @@ func ListApplications(key string) ([]Application, error) {
 	return applications, nil
 }
 
+var GetApplicationOptions = struct {
```
Comment (yesnault) [inline sdk/application.go]: doc?

File: sdk/exportentities/application.go [hunk 2]
```diff
@@ -0,0 +1,305 @@
+package exportentities
+
+import (
+	"text/template"
+
+	"github.com/ovh/cds/sdk"
+)
+
+type Application struct {
```
Comment (sguiheux) [inline sdk/exportentities/application.go]: doc

File: sdk/exportentities/application.go [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+	ProjectKey        string                         `json:"project_key" yaml:"project_key"`
+	Name              string                         `json:"name" yaml:"name"`
+	RepositoryManager string                         `json:"repo_manager,omitempty" yaml:"repo_manager,omitempty"`
+	RepositoryName    string                         `json:"repo_name,omitempty" yaml:"repo_name,omitempty"`
+	Permissions       map[string]int                 `json:"permissions,omitempty" yaml:"permissions,omitempty"`
+	Variables         map[string]VariableValue       `json:"variables,omitempty" yaml:"variables,omitempty"`
+	Pipelines         map[string]ApplicationPipeline `json:"pipelines,omitempty" yaml:"pipelines,omitempty"`
+}
+
+type ApplicationPipeline struct {
```
Comment (sguiheux) [inline sdk/exportentities/application.go]: doc

File: sdk/exportentities/application.go [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+	Parameters map[string]VariableValue              `json:"parameters,omitempty" yaml:"parameters,omitempty"`
+	Triggers   map[string]ApplicationPipelineTrigger `json:"triggers,omitempty" yaml:"triggers,omitempty"`
+	Options    []ApplicationPipelineOptions          `json:"options,omitempty" yaml:"options,omitempty"`
+}
+
+type ApplicationPipelineOptions struct {
```
Comment (sguiheux) [inline sdk/exportentities/application.go]: doc

File: sdk/exportentities/application.go [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+	Environment   *string                                    `json:"environment,omitempty" yaml:"environment,omitempty"`
+	Hook          *bool                                      `json:"hook,omitempty" yaml:"hook,omitempty"`
+	Polling       *bool                                      `json:"polling,omitempty" yaml:"polling,omitempty"`
+	Notifications map[string]ApplicationPipelineNotification `json:"notifications,omitempty" yaml:"notifications,omitempty"`
+	Schedulers    []ApplicationPipelineScheduler             `json:"schedulers,omitempty" yaml:"schedulers,omitempty"`
+}
+
+type ApplicationPipelineScheduler struct {
```
Comment (sguiheux) [inline sdk/exportentities/application.go]: doc

File: sdk/exportentities/application.go [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+	CronExpr   string                   `json:"cron_expr" yaml:"cron_expr"`
+	Parameters map[string]VariableValue `json:"parameters,omitempty" yaml:"parameters,omitempty"`
+}
+
+type ApplicationPipelineNotification sdk.UserNotificationSettings
```
Comment (sguiheux) [inline sdk/exportentities/application.go]: doc and everywhere below

File: sdk/exportentities/application.go [hunk 7]
```diff
@@ -0,0 +1,305 @@
...
+				Manual:          t.Manual,
+				Conditions:      c,
+			}
+		}
+
+		mapEnvOpts := map[string]*ApplicationPipelineOptions{}
+		//Hooks
+		for _, h := range app.Hooks {
+			if h.Enabled && h.Pipeline.Name == ap.Pipeline.Name {
+				if _, ok := mapEnvOpts["NoEnv"]; !ok {
```
Comment (sguiheux) [inline sdk/exportentities/application.go]: sdk.DefaultEnv.Name  here and everywhere

**Pull request merged**